### PR TITLE
updated new pipeline docs (issue #1615)

### PR DESCRIPTION
**Pull request by cezargrig98**
Edited broken link, file and folders in the template, and also created a link to a detailed guide for the modules section

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1727"></a>

### CODE REVIEW

File: markdown/developers/adding_pipelines.md [hunk 1]
```diff
@@ -299,6 +297,22 @@ Basic rules for such contributions:
...
+
+```bash
+process {
+    withName:foo {
+        container = 'image_name_1'
+    }
+    withName:bar {
+        container = 'image_name_2'
+    }
+    }
```
Comment (Emiller88) [inline markdown/developers/adding_pipelines.md:310]: Can't tell on mobile whether this is properly indented or not

File: markdown/developers/adding_pipelines.md [hunk 2]
```diff
@@ -299,6 +297,22 @@ Basic rules for such contributions:
...
+charliecloud {
+    enabled = true
+}
+```
```
Comment (Emiller88) [inline markdown/developers/adding_pipelines.md:314]: I think prettier wants it to have a space

**Review by Emiller88 — COMMENTED** [review]
Few things, I can't give this a more in-depth look tonight.

File: markdown/developers/adding_pipelines.md (hunk 2 above)
Comment (cezargrig98) [reply markdown/developers/adding_pipelines.md:314]: I swear to god prettier is giving me PTSD, i gotta bring it up with my therapist after this hackaton

**Comment by cezargrig98** [comment]
Btw, i couldn't fix the linting error on the troubleshooting doc because apparently Prettier considers that the wikipedia glob pattern link requires to be between <>, that's why formatting resulted in the link being broken

**Comment by mashehu** [comment]
@nf-core-bot fix linting

**Comment by cezargrig98** [comment]
@nf-core-bot fix linting

File: markdown/developers/adding_pipelines.md [hunk 3]
```diff
@@ -140,14 +138,14 @@ to the [repository GitHub web page](https://github.com/nf-core/test-datasets/) a
...
-([Slack](https://nf-co.re/join/slack) or [mailing list](https://groups.google.com/forum/#!forum/nf-core))
+Don't forget to ask for help if you have any doubts!
+([Slack](https://nf-co.re/join/slack),alternatively if you already joined Slack you can go directly to the [#help](https://nfcore.slack.com/channels/help) channel)
 
 ### Setting up a test workflow
 
 Now that your test data is hosted on the web, you can set up a `test` config profile in your
 workflow that points to it.
-In fact, the `test` profile should already exist if you've used the template.
+In fact, the `test` profile should already exist if you've used the template (you can find it in nf-core/my_workflow/conf/test.config).
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:148]:
```suggestion
A stub `test` profile should already exist in `conf/test.config`, so you just need to edit that file.
```

File: markdown/developers/adding_pipelines.md [hunk 4]
```diff
@@ -299,6 +297,24 @@ Basic rules for such contributions:
...
+    withName:foo {
+        container = 'image_name_1'
+        }
+    withName:bar {
+        container = 'image_name_2'
+        }
+      }
+ charliecloud {
+    enabled = true
+    }
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:313]:
```suggestion
process {
    withName:foo {
        container = 'image_name_1'
    }
    withName:bar {
        container = 'image_name_2'
    }
}
charliecloud {
    enabled = true
}
```

File: markdown/developers/adding_pipelines.md [hunk 5]
```diff
@@ -299,6 +297,24 @@ Basic rules for such contributions:
 Sometimes, especially when adding new features to a pipeline, the dependencies change as well. In such cases, you might want to have an updated Docker Container available before submitting a pull request, in order to have the GitHub Actions tests run through when testing your updated code. To achieve that, please follow these steps:
 
 - Add _only_ the newly required dependencies to the `environment.yml` in the pipeline code
+- If you only add new processes to an already existing pipeline however, you can simply specify the container in the `nextflow.config` file, like so:
+
+```bash
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:302]:
````suggestion
```nextflow
````

File: markdown/developers/adding_pipelines.md [hunk 6]
```diff
@@ -333,12 +349,22 @@ You will find the following files in each nf-core pipeline. They are automatical
...
+
+- `.prettierignore`: Files that should be ignored by prettier
 
 - `modules.json`: This file holds information (e.g. version) about all the modules in the pipeline that have been installed from `nf-core/modules`
 
+- `.nf-core.yml`: Indicates the type of repository (the standard being pipeline)
+
+- `.gitpod.yml`: Extensions to properly visualise the pipeline on github
+
+- `pyproject.toml`: Config file for Python. Mostly used to configure linting of bin/check_samplesheet.py with Black
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:362]:
```suggestion
- `pyproject.toml`: Config file for Python. Mostly used to configure linting of `bin/check_samplesheet.py` with Black
```

File: markdown/developers/adding_pipelines.md [hunk 7]
```diff
@@ -333,12 +349,22 @@ You will find the following files in each nf-core pipeline. They are automatical
...
-- `.markdownlint.yml`: Markdown lint configuration file to assure consistent markdown files
+- `.prettierrc.yml`: Prettier lint configuration file to assure consistent markdown files
+
+- `.prettierignore`: Files that should be ignored by prettier
 
 - `modules.json`: This file holds information (e.g. version) about all the modules in the pipeline that have been installed from `nf-core/modules`
 
+- `.nf-core.yml`: Indicates the type of repository (the standard being pipeline)
+
+- `.gitpod.yml`: Extensions to properly visualise the pipeline on github
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:360]:
```suggestion
- `.gitpod.yml`: Config file to define online working environment with <https://www.gitpod.io>
```

File: markdown/developers/adding_pipelines.md [hunk 8]
```diff
@@ -333,12 +349,22 @@ You will find the following files in each nf-core pipeline. They are automatical
...
 - `.editorconfig`: Editorconfig file that helps assuring consistent coding style
 
-- `.markdownlint.yml`: Markdown lint configuration file to assure consistent markdown files
+- `.prettierrc.yml`: Prettier lint configuration file to assure consistent markdown files
+
+- `.prettierignore`: Files that should be ignored by prettier
 
 - `modules.json`: This file holds information (e.g. version) about all the modules in the pipeline that have been installed from `nf-core/modules`
 
+- `.nf-core.yml`: Indicates the type of repository (the standard being pipeline)
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:358]:
```suggestion
- `.nf-core.yml`: Indicates the type of repository (pipeline or modules repo)
```

File: markdown/developers/adding_pipelines.md [hunk 9]
```diff
@@ -333,12 +349,22 @@ You will find the following files in each nf-core pipeline. They are automatical
...
+
+- `.gitpod.yml`: Extensions to properly visualise the pipeline on github
+
+- `pyproject.toml`: Config file for Python. Mostly used to configure linting of bin/check_samplesheet.py with Black
+
 ### Directories
 
+-`.devcontainer.json`: Specific container to facilitate updating the pipeline
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:366]:
```suggestion
- `.devcontainer`: Configuration to work with the [GitHub Codespaces](https://github.com/features/codespaces) online editing environments.
```

File: markdown/developers/adding_pipelines.md [hunk 10]
```diff
@@ -370,26 +396,13 @@ To assure that nf-core pipelines don't break after some change is made to the co
 
 ## DSL2 and modules
 
-Nextflow DSL2 allows for a more modularized design of pipelines and the reuse of components. Currently, most nf-core pipelines are still entirely written in DSL1, but in the near future all pipelines will be written in DSL2. The nf-core team has developed a set of design patterns on how to best implement DSL2 pipelines, which should be used by all nf-core pipelines in order to assure standardization and the reuse of components. The following is meant to help understand certain design choices and how a nf-core DSL2 pipeline should be build.
+Nextflow DSL2 allows for a more modularized design of pipelines and the reuse of components. Currently, most nf-core pipelines are written in DSL2, and DSL1 is considered outdated. The nf-core team has developed a set of design patterns on how to best implement DSL2 pipelines, which should be used by all nf-core pipelines in order to assure standardization and the reuse of components. The following is meant to help understand certain design choices and how a nf-core DSL2 pipeline should be build.
```
Comment (ewels) [inline markdown/developers/adding_pipelines.md:399]: No longer needed - DSL1 won't even run any more.

```suggestion
Nextflow DSL2 allows for a more modularized design of pipelines and the reuse of components. The nf-core team has developed a set of design patterns on how to best implement DSL2 pipelines, which should be used by all nf-core pipelines in order to assure standardization and the reuse of components. The following is meant to help understand certain design choices and how a nf-core DSL2 pipeline should be build.
```

File: markdown/usage/troubleshooting.md [hunk 11]
```diff
@@ -307,7 +307,7 @@ If you work in a `dev` branch, you may also want to consider putting a request f
 docker {
     enabled = true
     runOptions = '--pull=always'
-}
+    }
```
Comment (ewels) [inline markdown/usage/troubleshooting.md:310]:
```suggestion
}
```

**Review by ewels — APPROVED** [review]
Mostly good, made a few changes

**Pull request merged**